art/attacks/evasion/graphite/graphite_blackbox.py

```diff
@@ -0,0 +1,894 @@
+        :param off_x_range: The range of the x offset (percent) in the perspective transform.
+        :param off_y_range: The range of the y offset (percent) in the perspective transform.
+        :param blur_kernels: The kernels to blur with.
+        :param batch_size: The size of the batch used by the estimator during inference.
+        """
+        super().__init__(estimator=classifier)
+        self.noise_size = noise_size
+        self.net_size = net_size
+        self.heat_patch_size = heat_patch_size
+        self.heat_patch_stride = heat_patch_stride
+        self.heatmap_mode = heatmap_mode
+        self.batch_size = batch_size
+        self.tr_lo = tr_lo
+        self.tr_hi = tr_hi
+        self.num_xforms_mask = num_xforms_mask
+        self.max_mask_size = max_mask_size
+        self.beta = beta
+        self.eta = eta
+        self.num_xforms_boost = num_xforms_boost
+        self.num_boost_queries = num_boost_queries
+        self.rotation_range = rotation_range
+        self.dist_range = dist_range
+        self.gamma_range = gamma_range
+        self.crop_percent_range = crop_percent_range
+        self.off_x_range = off_x_range
+        self.off_y_range = off_y_range
+        self.blur_kernels = blur_kernels
+
+        self._check_params()
+
+    def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.ndarray:
+        """
+        Generate adversarial samples and return them in an array.
+
+        :param x: An array with the original inputs to be attacked.
+        :param y: Target values (class labels) one-hot-encoded of shape `(nb_samples, nb_classes)` or indices of shape
+                  (nb_samples,).
+        :param mask: An array with a mask broadcastable to input `x` defining where to apply adversarial perturbations.
+                     Shape needs to be broadcastable to the shape of x and can also be of the same shape as `x`. Any
+                     features for which the mask is zero will not be adversarially perturbed.
+        :param x_tar: Initial array to act as the example target image.
+        :param pts: Optional points to consider when cropping the perspective transform. An array of points in
+                    [x, y, scale] with shape [num points, 3, 1].
+        :param obj_width: The estimated object width (inches) for perspective transform. 30 by default.
+        :param focal: The estimated focal length (ft) for perspective transform. 3 by default.
+        :return: An array holding the adversarial examples.
+        """
+        mask = kwargs.get("mask")
+        x_tar = kwargs.get("x_tar")
+        obj_width = kwargs.get("obj_width") if "obj_width" in kwargs else 30
+        focal = kwargs.get("focal") if "focal" in kwargs else 3
+        pts = kwargs.get("pts") if "pts" in kwargs else None
+
+        if y is None:
+            raise ValueError("Target labels `y` need to be provided.")
+
+        assert isinstance(y, np.ndarray)
```

Comment:
if you want to type check, you should use an if not isinstance(): raise SpecificError(errormsg).